rpms/SPECS/4.0.0/wazuh-manager-4.0.0.spec

```
@@ -487,14 +484,13 @@ fi
 # posttrans code is the last thing executed in a install/upgrade
 %posttrans
 if [ -f %{_localstatedir}/tmp/wazuh.restart ]; then
-  if command -v systemctl > /dev/null 2>&1 && systemctl > /dev/null 2>&1 ; then
-    rm -f %{_localstatedir}/tmp/wazuh.restart
+  rm -f %{_localstatedir}/tmp/wazuh.restart
+  if command -v systemctl > /dev/null 2>&1 && systemctl > /dev/null 2>&1 && systemctl is-active --quiet wazuh-agent > /dev/null 2>&1; then
     systemctl restart wazuh-manager.service > /dev/null 2>&1
-  fi
-
-  if command -v service > /dev/null 2>&1; then
-    rm -f %{_localstatedir}/tmp/wazuh.restart
+  elif command -v service > /dev/null 2>&1 && service wazuh-agent status 2>/dev/null | grep "running" > /dev/null 2>&1; then
```

```suggestion
  elif command -v service > /dev/null 2>&1; then
```
At this point, the manager must be stopped.